File: backend/src/FormBuilder.Domain/Enums/ApprovalActionType.cs
namespace FormBuilder.Domain.Enums;

/// <summary>
/// The action an approver is allowed to take at a given step of an approval route.
/// </summary>
public enum ApprovalActionType
{
    Approve = 0,
    Reject = 1,
    ApproveOrReject = 2,
    ViewOnly = 3
}

File: frontend/src/app/core/models/approval-action-type.ts
/**
 * Actions an approver may take at a step of a form's approval route.
 * Values must match the backend's ApprovalActionType enum names exactly (serialized as strings).
 */
export enum ApprovalActionType {
  Approve = 'Approve',
  Reject = 'Reject',
  ApproveOrReject = 'ApproveOrReject',
  ViewOnly = 'ViewOnly',
}

export const APPROVAL_ACTION_LABELS: Record<ApprovalActionType, string> = {
  [ApprovalActionType.Approve]: 'אישור בלבד',
  [ApprovalActionType.Reject]: 'דחייה בלבד',
  [ApprovalActionType.ApproveOrReject]: 'אישור או דחייה',
  [ApprovalActionType.ViewOnly]: 'צפייה בלבד',
};
